tsmo/src/tsc_service_log_parser.py

```diff
@@ -0,0 +1,96 @@
+import sys
+from csv import writer
+import os
+import constants
+import json
+import pandas as pd
+import shutil
+
+from datetime import datetime, timezone
+import pytz
+import re
+#parser method to extract necessary fields from raw text file
+def serviceLogParser(logname):
+    input_directory_path = f'{constants.DATA_DIR}/{constants.RAW_LOG_DIR}'
+    output_directory_path = f'{constants.DATA_DIR}/{constants.PARSED_OUTPUT_DIR}'
+    all_in_filenames = os.listdir(input_directory_path)
+
+    for file in all_in_filenames:
+        if logname in file:
+            fileName = file.split("_")[2]
+            #Convert the text file into an array of lines
+            with open(f'{input_directory_path}/{file}', encoding="utf8", errors='ignore') as textFile:
+                textList = []
+                for line in textFile:
+                    textList.append(line.strip())
+
+            #write data of interest to csv which will be used to produce plots
+            print("Creating ", fileName + '_spat_processing_time.csv' )
+            with open(f'{output_directory_path}/{fileName}_spat_processing_time.csv', 'w', newline='') as write_obj:
+                csv_writer = writer(write_obj)
+                csv_writer.writerow(["Epoch_Time(ms)", "Processing_Time(ms)"])
+
+                #Need to get time since epoch of first day of year to use with moy and timestamp
```

Comment:
Can we remove this comment here and below, think this got copied over from a different file